embedding/cpp/include/flutter_engine.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+#define FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+
+#include <app_control.h>
+#include <flutter/plugin_registry.h>
+#include <flutter_tizen.h>
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+// The engine for flutter execution.
+class FlutterEngine : public flutter::PluginRegistry {
+ public:
+  virtual ~FlutterEngine();
+  static std::unique_ptr<FlutterEngine> Create(
```

Comment:
Private constructor and providing a creation method that returns pointer make that object can only be dynamically created.
also, if the return type is `unique_ptr`, it strongly suggests to the user that only one pointer must exist for instance.
so the method that takes this as an argument also uses `unique_ptr`. and as you can see from the code, it can return `nullptr` to the user if a just created instance is not valid.